dspace-server-webapp/src/main/java/org/dspace/app/rest/security/jwt/JWTTokenRestAuthenticationServiceImpl.java

```diff
@@ -125,7 +130,14 @@ public EPerson getAuthenticatedEPerson(HttpServletRequest request, HttpServletRe
                 token = getShortLivedToken(request);
                 ePerson = shortLivedJWTTokenHandler.parseEPersonFromToken(token, request, context);
             } else {
-                ePerson = loginJWTTokenHandler.parseEPersonFromToken(token, request, context);
+                if (ClarinTokenUtils.isClarinToken(token)) {
+                    ePerson = clarinTokenService.getEPersonFromClarinToken(context, token);
+                    if (ePerson != null) {
+                        context.setCurrentUser(ePerson);
```

Comment:
can we have a `contex.setAuthenticationMethod` as well (see https://github.com/ufal/clarin-dspace/blob/16e704a285f69ba2406791010db332decde7eb59/dspace-server-webapp/src/main/java/org/dspace/app/rest/security/jwt/AuthenticationMethodClaimProvider.java#L49)

otherwise /api/authn/status returns null:
```
{
  "id" : null,
  "okay" : true,
  "authenticated" : true,
  "authenticationMethod" : null,
```